### ISSUE TITLE
Default build task does not start

### ISSUE THREAD

**Issue by ivanvoznyakovsky** [microsoft/vscode#99797]
Issue Type: <b>Bug</b>

Configure default build task
Restart vscode
Press Shift+Cmd+B to run default build task
Observe nothing happens
Dev console outputs `ERR Task to execute is undefined`.
Same happens when starting build task via `Tasks: Run Build Task`

VS Code version: Code 1.46.0 (a5d1cc28bb5da32ec67e86cc50f84c67cc690321, 2020-06-10T08:59:06.977Z)
OS version: Darwin x64 19.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz (8 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.31GB free)|
|Process Argv|-psn_0_5539144|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|11.2.0
project-manager|ale|11.0.1
bracket-pair-colorizer-2|Coe|0.1.4
vscode-eslint|dba|2.1.5
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
vscode-todo-plus|fab|4.17.1
vscode-pull-request-github|Git|0.16.1
rainbow-csv|mec|1.7.0
git-graph|mhu|1.23.0
vsliveshare|ms-|1.0.2274
vscode-sort-json|ric|1.18.0
code-settings-sync|Sha|3.4.3
code-spell-checker|str|1.9.0
sort-lines|Tyr|1.9.0
vscodeintellicode|Vis|1.2.8
vscode-icons|vsc|10.1.1
vim|vsc|1.14.5
markdown-all-in-one|yzh|3.0.0

(1 theme extensions excluded)

</details>

**Comment by alexr00** [comment]
Can you provide a repository or a tasks.json I can test with? I cannot reproduce the issue with this task:
```
		{
			"label": "test",
			"command": "echo test",
			"type": "shell",
			"group": {
				"kind": "build",
				"isDefault": true
			}
		}
``
or with this task
```
		{
			"type": "npm",
			"script": "watch",
			"problemMatcher": "$tsc-watch",
			"isBackground": true,
			"presentation": {
				"reveal": "never"
			},
			"group": {
				"kind": "build",
				"isDefault": true
			}
		}
```

**Comment by ivanvoznyakovsky** [comment]
@alexr00 this is my `tasks.json`
```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
      {
        "label": "Dev Server",
        "type": "gulp",
        "task": "default",
        "group": {
          "kind": "build",
          "isDefault": true
        },
        "presentation": {
          "group": "dev"
        }
      }
    ]
  }
```

**Comment by ivanvoznyakovsky** [comment]
gulp tasks auto detection is on

**Comment by alexr00** [comment]
I still can't reproduce the issue. My steps:
1. Open a folder with a default gulp task in VS Code (I used the vscode repository).
2. Paste your task into the tasks.json and make sure there are no other tasks marked as default build.
3. Tasks: Run Build Task

Can you provide a repository where this issue happens?

**Comment by panost** [comment]
For me pressing Ctrl+Shift+B or selecting it from the Terminal menu not only it does not start building, it doesn't even saves the changes.

But if I choose the default task from the menu "Run task" and then "Continue without scanning the task output" then it not only builds, but the functionality of Ctrl+Shift+B is back

**Comment by MuTsunTsai** [comment]
I'm experiencing the same issue. Every time I restart VS Code, Ctrl + Shift + B won't work, until I click on "Configure Default Build Task" in the "Terminal" from menu bar, and then it will work as long as VS Code is open. But next time I restart VS Code I would have to do this again. I believe this began to happen only since the last upgrade.

**Comment by alexr00** [comment]
To verify:
- Open VS Code repository
- In the workspace settings, remove `"gulp.autoDetect": "off",`
- Open tasks.json and delete 
```
			"group": {
				"kind": "build",
				"isDefault": true
			},
```
- Run the configure default build task command. Select any gulp task that you don't mind running.
- Reload
- Run the Tasks: Run Build task command.
- Verify that the gulp task runs task runs.

**Comment by MuTsunTsai** [comment]
@alexr00 No this doesn't work for me; I still have to re-configure default build task every time I restart VS Code. I do use gulp though.

**Comment by alexr00** [comment]
This change will be in 1.46.1 and is in insiders in 1.47. We verify *before* we release to make sure it's fixed in the stable release (in this case 1.46.1). Have you tried it in insiders?

**Comment by MuTsunTsai** [comment]
@alexr00 I see; I suppose I will wait for the release of 1.46.1 then.
